tests/sim/conversion_proxy.rs

```diff
@@ -227,6 +227,65 @@ fn test_transfer_with_low_deposit() {
     );
 }
 
+#[test]
+fn test_transfer_high_amounts() {
```

Comment:
If we weren't in a rush, I'd say we need more tests.

Reply:
What do you have in mind? I can totally add more tests.

FYI I considered adding more runtime checks (that would need more tests), like a precision ceiling for example, and making sure that if you try to make a conversion that would be too imprecise, the transaction would `refund_then_log`. I did not prioritize this because I think a conversion rate with 9 digits is already huge (and managed without loss of precision), and you would have to have a conversion rate super precise, together with a very high payment amount, to lose relevant precision.

Reply:
I was thinking tests that show the limitations of the logic. 

Thank you for explaining some of the features you considered but didn't implement.

It's okay for now, this test is sufficient.